test/unit_tests/workflows/__init__.py

```diff
@@ -156,13 +156,21 @@ def __repr__(self) -> str:
 
 
 def _store_step(step_log: List[Tuple[Step, WFProcess]]) -> Callable[[ProcessStat, Step, WFProcess], WFProcess]:
-    def __store_step(pstat: ProcessStat, step: Step, state: WFProcess) -> WFProcess:
+    def __store_step(pstat: ProcessStat, step: Step, process: WFProcess) -> WFProcess:
         try:
-            state = state.map(lambda s: json_loads(json_dumps(s)))
+            process = process.map(lambda s: json_loads(json_dumps(s)))
         except Exception:
-            logger.exception("Step state is not valid json", state=state)
-        step_log.append((step, state))
-        return state
+            logger.exception("Step state is not valid json", process=process)
+
+        state = process.unwrap()
+        state.pop("__step_name_override", None)
+        for k in state.get("__remove_keys", []) + ["__remove_keys"]:
+            state.pop(k, None)
+        if state.pop("__replace_last_state", None):
+            step_log[-1] = (step, process)
+        else:
+            step_log.append((step, process))
+        return process
```

Comment:
This looks very similar to the actual code, and I assume we have to keep it carefully in sync? Would it make sense to extract this as a function, so it can be reused here?